src/main.rs

```diff
@@ -279,6 +279,7 @@ fn run() -> Result<ExitStatus> {
 
     let host_version_meta =
         rustc_version::version_meta().wrap_err("couldn't fetch the `rustc` version")?;
+    let is_nightly = rustc::is_nightly(&args.channel, &host_version_meta);
```

Comment:
this is wrong, it should check the target meta, not the host meta. see the logic in L325/L324

Reply:
I believe this is fixed by detecting the channel in `rustup::rustc_version`. First we detect if the toolchain contains nightly, and if we can parse the toolchain info via `rustup::rustc_version`, then we update that. This might be slightly redundant, however: I'm not sure if I need both assignments. The logic is effectively:

```rust
let toolchain = ...;
let mut is_nightly = toolchain.contains("nightly");
if let Some((rustc_version, channel, rustc_commit)) = rustup::rustc_version(&sysroot)? {
    is_nightly = channel == Channel::Nightly;
}
```